src/components/Clips.js

```diff
@@ -179,16 +186,53 @@ class Clips extends React.Component {
     this.setState({modal: null});
   }
 
+  ClipsList() {
+    let clips;
+    if(this.props.playlistIndex !== undefined) {
+      clips = this.props.formStore.currentLocalizedData.playlists[this.props.playlistIndex].clips;
+    } else {
+      clips = this.props.formStore.currentLocalizedData.assets[this.props.storeKey];
+    }
+
+    if(this.state.activeFilter) {
+      clips = clips.filter(clip => clip.displayTitle.includes(this.state.activeFilter));
+    }
+
+    return clips;
+  }
+
   render() {
-    const clips = this.props.playlistIndex !== undefined ?
-      this.props.formStore.currentLocalizedData.playlists[this.props.playlistIndex].clips :
-      this.props.formStore.currentLocalizedData.assets[this.props.storeKey];
+    const clips = this.ClipsList();
 
     return (
       <div className="asset-form-section-container">
         <h3>{this.props.header}</h3>
+        <div className="controls">
+          <Action onClick={this.ActivateModal}>
+            Add {this.props.name}
+          </Action>
+          { this.Filter("Filter Titles...") }
+        </div>
+        { this.PageControls(clips.length) }
         <div className="asset-form-clips-container">
-          {(clips || []).map((clip, index) =>
+          {
+            clips.length ? <div
+              className={`
+            asset-form-clip 
+            ${this.props.orderable   ? "asset-form-clip-orderable" : ""}
+            ${this.props.defaultable ? "asset-form-clip-defaultable" : ""}
+            ${this.props.showPreview ? "asset-form-clip-with-preview" : ""}
+          `}>
+              <div></div>
+              <div>Type</div>
```

Comment:
Please make the column headers sortable (modify sortKey and sortAsc in the component state). Don't forget to add cursor: pointer to clickable headers